components/automate-cs-ocid/habitat/config/tasks/oauth_application.rake

```diff
@@ -0,0 +1,37 @@
+require 'active_record/base'
+require 'fileutils'
+namespace :oauth_application do
+  desc "Task to register oauth applications"
+  task :register => :environment do
+    # This task expects an environment variable to be set with a json string
+    # containing a list of oauth application objects. The list should have the following schema:
+    # [{"name"=>"<app name>", "redirect_uri"=>"<callback url for redirect>"}]
+    oauth_applications_json = ENV['OAUTH_APPLICATIONS_JSON']
+    oauth_applications = JSON.parse(oauth_applications_json)
+    ActiveRecord::Base.transaction do
+      oauth_applications.each do |oauth_application|
+        # Making sure the iteration doesn't execute in case of empty value for app name
+        next if oauth_application['name'].blank?
+        app = Doorkeeper::Application.find_or_create_by(:name => oauth_application['name'])
+        # Updating redirect_uri of the app record only if it's different
+        # from the existing value in case it's an existing record
+        app.update!(:redirect_uri => oauth_application['redirect_uri']) if (app.redirect_uri != oauth_application['redirect_uri'])
+      end
+    end
+  end
+
+  desc "Task to generate a file with the details of all the registered oauth application under OC-ID"
+  task :save_registered_app_details_to_file => :environment do
+    registered_apps = Doorkeeper::Application.select(:name, :redirect_uri, :uid, :secret).group_by(&:name).as_json
+    yaml_file_content = registered_apps.to_yaml
+    file_path = ENV['REGISTERED_OAUTH_APPS_FILE_PATH']
+
+    # Making sure the directory is created before writing in the file
+    dir = File.dirname(file_path)
+    unless File.directory?(dir)
+      FileUtils.mkdir_p(dir)
+    end
+    
+    File.write(file_path, yaml_file_content)
```

Comment:
Make sure any error in file operations is rescued and the error is reported without breaking the flow as this is a non critical task for the runtime of OCID.